perfmetrics/scripts/ls_metrics/listing_benchmark_test.py

```diff
@@ -0,0 +1,519 @@
+
+    @patch('listing_benchmark._list_directory')
+    def test_compare_directory_structure_false_file_type2(self, mock_list):
+        mock_list.side_effect = [
+            ['fake_bucket/', 'fake_bucket/2KB_3files_0subdir/',
+                'fake_bucket/1KB_2files_0subdir/', 'fake_bucket/1KB_0files_0subdir/'],
+            ['fake_bucket/2KB_3files_0subdir/file_1', 'fake_bucket/2KB_3files_0subdir/file_2',
+                'fake_bucket/2KB_3files_0subdir/file_3'],
+            ['fake_bucket/1KB_2files_0subdir/file_1',
+                'fake_bucket/1KB_2files_0subdir/file_2'],
+            []
+        ]
+        result = listing_benchmark._compare_directory_structure(
+            'fake_bucket/', DIRECTORY_STRUCTURE2)
+        self.assertFalse(result)
+
+    @patch('listing_benchmark._list_directory')
+    def test_compare_directory_structure_true_type3(self, mock_list):
+        mock_list.side_effect = [
+            ['fake_bucket/', 'fake_bucket/1KB_4files_3subdir/',
+                'fake_bucket/2KB_3files_1subdir/', 'fake_bucket/1KB_1files_0subdir/'],
+            ['fake_bucket/1KB_4files_3subdir/file_1', 'fake_bucket/1KB_4files_3subdir/file_2', 'fake_bucket/1KB_4files_3subdir/file_3', 'fake_bucket/1KB_4files_3subdir/file_4',
+             'fake_bucket/1KB_4files_3subdir/subdir1/', 'fake_bucket/1KB_4files_3subdir/subdir2/', 'fake_bucket/1KB_4files_3subdir/subdir3/'],
+            ['fake_bucket/1KB_4files_3subdir/subdir1/file_1', 'fake_bucket/1KB_4files_3subdir/subdir1/subsubdir1/',
+                'fake_bucket/1KB_4files_3subdir/subdir1/subsubdir2/'],
+            ['fake_bucket/1KB_4files_3subdir/subdir1/subsubdir1/file_1',
+                'fake_bucket/1KB_4files_3subdir/subdir1/subsubdir1/file_2'],
+            [],
+            ['fake_bucket/1KB_4files_3subdir/subdir2/file_1'],
+            ['fake_bucket/1KB_4files_3subdir/subdir3/file_1',
+                'fake_bucket/1KB_4files_3subdir/subdir3/subsubdir1/'],
+            ['fake_bucket/1KB_4files_3subdir/subdir3/subsubdir1/file_1'],
+            ['fake_bucket/2KB_3files_1subdir/file_1', 'fake_bucket/2KB_3files_1subdir/file_2',
+                'fake_bucket/2KB_3files_1subdir/file_3', 'fake_bucket/2KB_3files_1subdir/subdir1/'],
+            [],
+            ['fake_bucket/1KB_1files_0subdir/file_1']
+        ]
+        result = listing_benchmark._compare_directory_structure(
+            'fake_bucket/', DIRECTORY_STRUCTURE3)
+        self.assertTrue(result)
+
+    @patch('listing_benchmark._list_directory')
+    def test_compare_directory_structure_false_file_folder_type3(self, mock_list):
+        mock_list.side_effect = [
+            ['fake_bucket/', 'fake_bucket/file1', 'fake_bucket/1KB_4files_3subdir/',
+                'fake_bucket/2KB_3files_1subdir/', 'fake_bucket/1KB_1files_0subdir/'],
+            ['fake_bucket/1KB_4files_3subdir/file_1', 'fake_bucket/1KB_4files_3subdir/file_2', 'fake_bucket/1KB_4files_3subdir/file_3', 'fake_bucket/1KB_4files_3subdir/file_4',
```

Comment:
formatting here and at other lines which exceed 80 chars. Please follow google python guidelines here

Reply:
changed....but some lines are still slightly greater than 80 because they contain a path url or linux command.